### DISCUSSION TITLE
put duplicate

### DISCUSSION THREAD

**Discussion by cekvenich2** [kriszyp/lmdb-js#120] [Q&A] [answered]
Hi, I have a question on what happens on put when the key is duplicate.
I'm clear on what happens when dupSort is true.

But when it's false, and I do a put, it appends anyway.
(Should I always check for a put with a get to avoid collision?)
tia

**Comment by kriszyp** [comment, marked as answer]
Doing a `put` with `dupSort: false` with an entry that already exists will replace it. So:
```
db = open(...); // no dupSort
db.put('key', 'first value');
db.put('key', 'second value'); // replaces the value for 'key'
await db;
db.get('key'); // returns 'second value' 
```
LMDB does have an option for doing `put`s that will only execute if there is no existing entry and fail if there is an existing entry:
```
db.put('key', 'second value', { noOverwrite: true });
// or
db.ifNoExists('key', () => db.put('key', 'second value'));
```
Both of these will (return a promise that) resolves to false if there is already an existing entry.

**Reply by cekvenich2** [reply]
Thx. Whatever issue I had I now can't reproduce.
